contracts/v2/sovereignChains/BridgeL2SovereignChain.sol

```diff
@@ -0,0 +1,393 @@
+// SPDX-License-Identifier: AGPL-3.0
+
+pragma solidity 0.8.20;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import "../../lib/TokenWrapped.sol";
+import "../../interfaces/IBasePolygonZkEVMGlobalExitRoot.sol";
+import "../interfaces/IBridgeL2SovereignChains.sol";
+import "../PolygonZkEVMBridgeV2.sol";
+
+/**
+ * Sovereign chains bridge that will be deployed on Ethereum and all Sovereign chains
+ * Contract responsible to manage the token interactions with other networks
+ */
+contract BridgeL2SovereignChain is
+    PolygonZkEVMBridgeV2,
+    IBridgeL2SovereignChains
+{
+    using SafeERC20Upgradeable for IERC20Upgradeable;
+
+    // Wrapped Token information struct
+    struct SovereignTokenAddress {
```

Comment:
i think this struct is overkill since it's just used in the loop call

Reply:
I guess it adds more bytecode to the SC and more gas consumption. I'm not sure if we can afford this additional cost Vs code-readability.
There are two approaches if we decide to not go with struct that I can think of:
- `bytes` (and do the decoding on the SC)
- `bytes[]` --> so each element is a packed version of the struct
- 4 arrays --> each for each element (sanity check should be done on the length, so all has the same)

Reply:
i would do 4 arrays ^^

Reply:
I finally used the struct as it consumes less bytecode of the contract and we are at the limit